pages/signin.tsx

```diff
@@ -1,16 +1,43 @@
-import { WithPageLayout } from "../interfaces/with-page-layout";
+import { useEffect } from "react";
+import { useRouter } from "next/router";
 
-const SignIn: WithPageLayout = () => {
-  return <main className="flex w-full flex-1 flex-col items-center justify-center px-20 text-center">...Loading</main>;
-};
+import { WithPageLayout } from "interfaces/with-page-layout";
+import LoginLayout from "layouts/login";
+import useSupabaseAuth from "lib/hooks/useSupabaseAuth";
 
-export const getServerSideProps = async () => {
-  return {
-    redirect: {
-      destination: "/",
-      permanent: false,
-    },
+const SignInPage: WithPageLayout = () => {
+  const { signIn } = useSupabaseAuth();
+  const router = useRouter();
+  const {
+    redirectedFrom = "/hub/insights/new",
+    destination,
+    auth,
+  } = router.query as {
+    redirectedFrom?: string;
+    destination?: string;
+    auth?: string;
   };
+  const hasParams = Object.keys(router.query).length > 0;
+  const hasAuth = auth === "true";
+
+  useEffect(() => {
+    if (hasParams) {
+      if (destination && hasAuth) {
+        router.replace(decodeURIComponent(destination));
```

Comment:
refactor (blocking): `router.query` already decodes the querystring values destructured above. Same for the `redirectedFrom` below.

```suggestion
router.replace(destination);
```